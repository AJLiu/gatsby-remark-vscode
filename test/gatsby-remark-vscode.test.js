// @ts-check
jest.mock('request');
jest.mock('decompress');
jest.mock('../src/utils');
const request = require('request');
const decompress = require('decompress');
const path = require('path');
const createPlugin = require('../src');
const utils = require('../src/utils');
const realUtils = jest.requireActual('../src/utils');

// @ts-ignore
utils.parseExtensionIdentifier.mockImplementation(realUtils.parseExtensionIdentifier);
// @ts-ignore
utils.getExtensionBasePath.mockImplementation(realUtils.getExtensionBasePath);
// @ts-ignore
utils.getExtensionPath.mockImplementation(realUtils.getExtensionPath);
// @ts-ignore
utils.getLanguageNames.mockImplementation(realUtils.getLanguageNames);
// @ts-ignore
utils.requireJson.mockImplementation(realUtils.requireJson);
// @ts-ignore
utils.requireGrammar.mockImplementation(realUtils.requireGrammar);

const markdownNode = { fileAbsolutePath: path.join(__dirname, 'test.md') };
/** @type {import('../src').PluginOptions} */
const options = { injectStyles: false };

function createCache() {
  return new Map();
}

function createMarkdownAST(lang = 'js', value = 'const x = 3;\n// Comment') {
  return {
    children: [
      { type: 'paragraph' },
      { type: 'code', lang, value },
      { type: 'paragraph' },
    ],
  };
}

describe('included languages and themes', () => {
  it('works with default options', async () => {
    const plugin = createPlugin();
    const markdownAST = createMarkdownAST();
    const cache = createCache();
    await plugin({ markdownAST, markdownNode, cache }, options);
    expect(markdownAST).toMatchSnapshot();
  });

  it('works without highlighting if language is not recognized', async () => {
    const plugin = createPlugin();
    const markdownAST = createMarkdownAST('none');
    const cache = createCache();
    await plugin({ markdownAST, markdownNode, cache }, options);
    expect(markdownAST).toMatchSnapshot();
  });

  it('works without highlighting if a code fence has no language', async () => {
    const plugin = createPlugin();
    const markdownAST = createMarkdownAST('');
    const cache = createCache();
    await plugin({ markdownAST, markdownNode, cache }, options);
    expect(markdownAST).toMatchSnapshot();
  });

  it('only adds theme CSS once', async () => {
    const plugin = createPlugin();
    const markdownAST = { children: [...createMarkdownAST().children, ...createMarkdownAST().children] };
    const cache = createCache();
    await plugin({ markdownAST, markdownNode, cache }, options);
    expect(markdownAST.children.filter(node => node.type === 'html')).toHaveLength(3);
  });

  it('can use a standard language alias', async () => {
    const plugin = createPlugin();
    const markdownAST = createMarkdownAST('JavaScript');
    const cache = createCache();
    await plugin({ markdownAST, markdownNode, cache }, options);
    expect(markdownAST).toMatchSnapshot();
  });

  it('can use a custom language alias', async () => {
    const plugin = createPlugin();
    const markdownAST = createMarkdownAST('java-scripty');
    const cache = createCache();
    await plugin({ markdownAST, markdownNode, cache }, { ...options, languageAliases: { 'java-scripty': 'js' } });
    expect(markdownAST).toMatchSnapshot();
  });
});

describe('extension downloading', () => {
  afterEach(() => {
    // @ts-ignore
    request.get.mockReset();
    // @ts-ignore
    decompress.mockReset();
  });

  it('can download an extension to resolve a theme', async () => {
    const plugin = createPlugin();
    // @ts-ignore
    const requestMock = request.get.mockImplementation((_, __, cb) => {
      cb(null, { statusCode: 200, headers: {} }, Buffer.from(''));
    });
    // @ts-ignore
    const decompressMock = decompress.mockImplementation(jest.fn());
    // @ts-ignore
    utils.requireJson.mockImplementation((jsonPath) => ({
      contributes: {
        themes: [{
          id: jsonPath.includes('wrong-one') ? 'wrong-one' : 'custom',
          label: jsonPath.includes('wrong-one') ? 'wrong-one' : 'Custom Theme',
          path: `../../../../test/${jsonPath.includes('wrong-one') ? 'wrong-one' : 'custom'}.tmTheme.json`
        }],
      }
    }));

    const markdownAST = createMarkdownAST();
    const cache = createCache();
    await plugin({ markdownAST, markdownNode, cache }, {
      ...options,
      colorTheme: 'Custom Theme',
      extensions: [{
        identifier: 'publisher.wrong-one',
        version: '1.0.0',
      }, {
        identifier: 'publisher.custom-theme',
        version: '1.0.0',
      }],
    });

    expect(requestMock).toHaveBeenCalledTimes(2);
    expect(decompressMock).toHaveBeenCalledTimes(2);
    expect(markdownAST).toMatchSnapshot();
  });

  it('can download an extension to resolve a grammar', async () => {
    const plugin = createPlugin();
    // @ts-ignore
    const requestMock = request.get.mockImplementation((_, __, cb) => {
      cb(null, { statusCode: 200, headers: {} }, Buffer.from(''));
    });
    // @ts-ignore
    const decompressMock = decompress.mockImplementation(jest.fn());
    // @ts-ignore
    utils.requireJson.mockImplementation((jsonPath) => ({
      contributes: {
        languages: [{
          id: jsonPath.includes('wrong-one') ? 'wrong-one' : 'custom',
        }],
        grammars: [{
          language: jsonPath.includes('wrong-one') ? 'wrong-one' : 'custom',
          scopeName: jsonPath.includes('wrong-one') ? 'source.wrong' : 'source.custom',
          path: `../../../../test/${jsonPath.includes('wrong-one') ? 'wrong-one' : 'custom'}.tmLanguage.json`
        }],
      }
    }));

    const markdownAST = createMarkdownAST('custom');
    const cache = createCache();
    await plugin({ markdownAST, markdownNode, cache }, {
      ...options,
      extensions: [{
        identifier: 'publisher.wrong-one',
        version: '1.0.0',
      }, {
        identifier: 'publisher.custom-language',
        version: '1.0.0',
      }],
    });

    expect(requestMock).toHaveBeenCalledTimes(2);
    expect(decompressMock).toHaveBeenCalledTimes(2);
    expect(markdownAST).toMatchSnapshot();
  });
});

it('sets highlighted line class names', async () => {
  const plugin = createPlugin();
  const markdownAST = createMarkdownAST('js{1,3-4}', '// 1\n// 2\n// 3\n// 4\n// 5');
  const cache = createCache();
  await plugin({ markdownAST, markdownNode, cache }, options);
  expect(markdownAST).toMatchSnapshot();
});

it('can replace a color value', async () => {
  const plugin = createPlugin();
  const markdownAST = createMarkdownAST();
  const cache = createCache();
  await plugin({ markdownAST, markdownNode, cache }, {
    ...options,
    replaceColor: oldColor => `var(--color-${oldColor.replace('#', '')})`,
  });
  expect(markdownAST).toMatchSnapshot();
});
