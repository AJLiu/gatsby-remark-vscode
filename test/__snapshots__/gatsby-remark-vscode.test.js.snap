// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can replace a color value 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "lang": "js",
      "type": "html",
      "value": "<pre class=\\"grvsc-container default-dark\\" data-language=\\"js\\"><code class=\\"grvsc-code\\"><span class=\\"grvsc-line\\"><span class=\\"mtk4\\">const</span><span class=\\"mtk1\\"> </span><span class=\\"mtk12\\">x</span><span class=\\"mtk1\\"> = </span><span class=\\"mtk7\\">3</span><span class=\\"mtk1\\">;</span></span>
<span class=\\"grvsc-line\\"><span class=\\"mtk3\\">// Comment</span></span></code></pre>",
    },
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "type": "html",
      "value": "<style class=\\"grvsc-styles\\">
  
  .default-dark {
    background-color: #1E1E1E;
    color: #D4D4D4;
  }
  .default-dark .mtk4 { color: var(--color-569CD6); }
  .default-dark .mtk1 { color: var(--color-D4D4D4); }
  .default-dark .mtk12 { color: var(--color-9CDCFE); }
  .default-dark .mtk7 { color: var(--color-B5CEA8); }
  .default-dark .mtk3 { color: var(--color-6A9955); }
</style>",
    },
  ],
  "type": "root",
}
`;

exports[`extension downloading can download an extension to resolve a grammar 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "lang": "custom",
      "type": "html",
      "value": "<pre class=\\"grvsc-container default-dark\\" data-language=\\"custom\\"><code class=\\"grvsc-code\\"><span class=\\"grvsc-line\\"><span class=\\"mtk1\\">const x = 3;</span></span>
<span class=\\"grvsc-line\\"><span class=\\"mtk1\\">// Comment</span></span></code></pre>",
    },
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "type": "html",
      "value": "<style class=\\"grvsc-styles\\">
  
  .default-dark {
    background-color: #1E1E1E;
    color: #D4D4D4;
  }
  .default-dark .mtk1 { color: #D4D4D4; }
</style>",
    },
  ],
  "type": "root",
}
`;

exports[`extension downloading can download an extension to resolve a theme 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "lang": "js",
      "type": "html",
      "value": "<pre class=\\"grvsc-container custom-theme\\" data-language=\\"js\\"><code class=\\"grvsc-code\\"><span class=\\"grvsc-line\\"><span class=\\"mtk4\\">const</span><span class=\\"mtk1\\"> x </span><span class=\\"mtk11\\">=</span><span class=\\"mtk1\\"> </span><span class=\\"mtk5\\">3</span><span class=\\"mtk1\\">;</span></span>
<span class=\\"grvsc-line\\"><span class=\\"mtk3\\">// Comment</span></span></code></pre>",
    },
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "type": "html",
      "value": "<style class=\\"grvsc-styles\\">
  
  .custom-theme .mtk4 { color: #569CD6; }
  .custom-theme .mtk1 { color: #000000; }
  .custom-theme .mtk11 { color: #D4D4D4; }
  .custom-theme .mtk5 { color: #B5CEA8; }
  .custom-theme .mtk3 { color: #6A9955; }
</style>",
    },
  ],
  "type": "root",
}
`;

exports[`included languages and themes can use a custom language alias 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "lang": "java-scripty",
      "type": "html",
      "value": "<pre class=\\"grvsc-container default-dark\\" data-language=\\"java-scripty\\"><code class=\\"grvsc-code\\"><span class=\\"grvsc-line\\"><span class=\\"mtk4\\">const</span><span class=\\"mtk1\\"> </span><span class=\\"mtk12\\">x</span><span class=\\"mtk1\\"> = </span><span class=\\"mtk7\\">3</span><span class=\\"mtk1\\">;</span></span>
<span class=\\"grvsc-line\\"><span class=\\"mtk3\\">// Comment</span></span></code></pre>",
    },
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "type": "html",
      "value": "<style class=\\"grvsc-styles\\">
  
  .default-dark {
    background-color: #1E1E1E;
    color: #D4D4D4;
  }
  .default-dark .mtk4 { color: #569CD6; }
  .default-dark .mtk1 { color: #D4D4D4; }
  .default-dark .mtk12 { color: #9CDCFE; }
  .default-dark .mtk7 { color: #B5CEA8; }
  .default-dark .mtk3 { color: #6A9955; }
</style>",
    },
  ],
  "type": "root",
}
`;

exports[`included languages and themes can use a standard language alias 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "lang": "JavaScript",
      "type": "html",
      "value": "<pre class=\\"grvsc-container default-dark\\" data-language=\\"javascript\\"><code class=\\"grvsc-code\\"><span class=\\"grvsc-line\\"><span class=\\"mtk4\\">const</span><span class=\\"mtk1\\"> </span><span class=\\"mtk12\\">x</span><span class=\\"mtk1\\"> = </span><span class=\\"mtk7\\">3</span><span class=\\"mtk1\\">;</span></span>
<span class=\\"grvsc-line\\"><span class=\\"mtk3\\">// Comment</span></span></code></pre>",
    },
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "type": "html",
      "value": "<style class=\\"grvsc-styles\\">
  
  .default-dark {
    background-color: #1E1E1E;
    color: #D4D4D4;
  }
  .default-dark .mtk4 { color: #569CD6; }
  .default-dark .mtk1 { color: #D4D4D4; }
  .default-dark .mtk12 { color: #9CDCFE; }
  .default-dark .mtk7 { color: #B5CEA8; }
  .default-dark .mtk3 { color: #6A9955; }
</style>",
    },
  ],
  "type": "root",
}
`;

exports[`included languages and themes includes special characters in language name 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "lang": "c++",
      "type": "html",
      "value": "<pre class=\\"grvsc-container default-dark\\" data-language=\\"c++\\"><code class=\\"grvsc-code\\"><span class=\\"grvsc-line\\"><span class=\\"mtk4\\">const</span><span class=\\"mtk1\\"> x = </span><span class=\\"mtk7\\">3</span><span class=\\"mtk1\\">;</span></span>
<span class=\\"grvsc-line\\"><span class=\\"mtk3\\">// Comment</span></span></code></pre>",
    },
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "type": "html",
      "value": "<style class=\\"grvsc-styles\\">
  
  .default-dark {
    background-color: #1E1E1E;
    color: #D4D4D4;
  }
  .default-dark .mtk4 { color: #569CD6; }
  .default-dark .mtk1 { color: #D4D4D4; }
  .default-dark .mtk7 { color: #B5CEA8; }
  .default-dark .mtk3 { color: #6A9955; }
</style>",
    },
  ],
  "type": "root",
}
`;

exports[`included languages and themes partially works if an embedded grammar is missing 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "lang": "embedded",
      "type": "html",
      "value": "<pre class=\\"default-dark vscode-highlight\\" data-language=\\"embedded\\"><code class=\\"vscode-highlight-code\\"><span class=\\"vscode-highlight-line\\">const x = 3;</span>
<span class=\\"vscode-highlight-line\\">// Comment</span></code></pre>",
    },
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "type": "html",
      "value": "<style class=\\"vscode-highlight-styles\\">
  
  .default-dark {
background-color: #1E1E1E;
color: #D4D4D4;
}

.default-dark .mtk1 { color: #D4D4D4; }
.default-dark .mtk2 { color: #1E1E1E; }
.default-dark .mtk3 { color: #6A9955; }
.default-dark .mtk4 { color: #569CD6; }
.default-dark .mtk5 { color: #D16969; }
.default-dark .mtk6 { color: #D7BA7D; }
.default-dark .mtk7 { color: #B5CEA8; }
.default-dark .mtk8 { color: #CE9178; }
.default-dark .mtk9 { color: #646695; }
.default-dark .mtk10 { color: #4EC9B0; }
.default-dark .mtk11 { color: #DCDCAA; }
.default-dark .mtk12 { color: #9CDCFE; }
.default-dark .mtk13 { color: #000080; }
.default-dark .mtk14 { color: #F44747; }
.default-dark .mtk15 { color: #C586C0; }
.default-dark .mtk16 { color: #6796E6; }
.default-dark .mtk17 { color: #808080; }
.default-dark .mtki { font-style: italic; }
.default-dark .mtkb { font-weight: bold; }
.default-dark .mtku { text-decoration: underline; text-underline-position: under; }
</style>",
    },
  ],
  "type": "root",
}
`;

exports[`included languages and themes works with default options 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "lang": "js",
      "type": "html",
      "value": "<pre class=\\"grvsc-container default-dark\\" data-language=\\"js\\"><code class=\\"grvsc-code\\"><span class=\\"grvsc-line\\"><span class=\\"mtk4\\">const</span><span class=\\"mtk1\\"> </span><span class=\\"mtk12\\">x</span><span class=\\"mtk1\\"> = </span><span class=\\"mtk7\\">3</span><span class=\\"mtk1\\">;</span></span>
<span class=\\"grvsc-line\\"><span class=\\"mtk3\\">// Comment</span></span></code></pre>",
    },
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "type": "html",
      "value": "<style class=\\"grvsc-styles\\">
  
  .default-dark {
    background-color: #1E1E1E;
    color: #D4D4D4;
  }
  .default-dark .mtk4 { color: #569CD6; }
  .default-dark .mtk1 { color: #D4D4D4; }
  .default-dark .mtk12 { color: #9CDCFE; }
  .default-dark .mtk7 { color: #B5CEA8; }
  .default-dark .mtk3 { color: #6A9955; }
</style>",
    },
  ],
  "type": "root",
}
`;

exports[`included languages and themes works without highlighting if a code fence has no language 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "lang": "",
      "type": "html",
      "value": "<pre class=\\"default-dark vscode-highlight\\" data-language=\\"\\"><code class=\\"vscode-highlight-code\\"><span class=\\"vscode-highlight-line\\">const x = 3;</span>
<span class=\\"vscode-highlight-line\\">// Comment</span></code></pre>",
    },
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "type": "html",
      "value": "<style class=\\"vscode-highlight-styles\\">
  
  .default-dark {
background-color: #1E1E1E;
color: #D4D4D4;
}
</style>",
    },
  ],
  "type": "root",
}
`;

exports[`included languages and themes works without highlighting if language is not recognized 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "lang": "none",
      "type": "html",
      "value": "<pre class=\\"default-dark vscode-highlight\\" data-language=\\"none\\"><code class=\\"vscode-highlight-code\\"><span class=\\"vscode-highlight-line\\">const x = 3;</span>
<span class=\\"vscode-highlight-line\\">// Comment</span></code></pre>",
    },
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "type": "html",
      "value": "<style class=\\"vscode-highlight-styles\\">
  
  .default-dark {
background-color: #1E1E1E;
color: #D4D4D4;
}
</style>",
    },
  ],
  "type": "root",
}
`;

exports[`prefers-color-scheme supports prefers-color-scheme via an object for \`colorTheme\` 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "lang": "js",
      "type": "html",
      "value": "<pre class=\\"grvsc-container solarized-light grvsc-mm-tuw09S\\" data-language=\\"js\\"><code class=\\"grvsc-code\\"><span class=\\"grvsc-line\\"><span class=\\"grvsc-tiaGIi-13,grvsc-tiaGIi-b grvsc-tuw09S-11\\">const</span><span class=\\"grvsc-tiaGIi-1 grvsc-tuw09S-1\\"> </span><span class=\\"grvsc-tiaGIi-7 grvsc-tuw09S-3\\">x</span><span class=\\"grvsc-tiaGIi-1 grvsc-tuw09S-1\\"> </span><span class=\\"grvsc-tiaGIi-8 grvsc-tuw09S-15\\">=</span><span class=\\"grvsc-tiaGIi-1 grvsc-tuw09S-1\\"> </span><span class=\\"grvsc-tiaGIi-6 grvsc-tuw09S-3\\">3</span><span class=\\"grvsc-tiaGIi-1 grvsc-tuw09S-1\\">;</span></span>
<span class=\\"grvsc-line\\"><span class=\\"grvsc-tiaGIi-3,grvsc-tiaGIi-i grvsc-tuw09S-4\\">// Comment</span></span></code></pre>",
    },
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "type": "html",
      "value": "<style class=\\"grvsc-styles\\">
  
  .solarized-light { background-color: #FDF6E3; }
  .solarized-light .grvsc-tiaGIi-i { font-style: italic; }
  .solarized-light .grvsc-tiaGIi-b { font-weight: bold; }
  .solarized-light .grvsc-tiaGIi-13 { color: #073642; }
  .solarized-light .grvsc-tiaGIi-1 { color: #657B83; }
  .solarized-light .grvsc-tiaGIi-7 { color: #268BD2; }
  .solarized-light .grvsc-tiaGIi-8 { color: #859900; }
  .solarized-light .grvsc-tiaGIi-6 { color: #D33682; }
  .solarized-light .grvsc-tiaGIi-3 { color: #93A1A1; }
  
  /* Monokai Dimmed */
  @media (prefers-color-scheme: dark) {
    .grvsc-mm-tuw09S {
      background-color: #1e1e1e;
      color: #c5c8c6;
    }
    .grvsc-mm-tuw09S .grvsc-tuw09S-11 { color: #9872A2; }
    .grvsc-mm-tuw09S .grvsc-tuw09S-1 { color: #C5C8C6FF; }
    .grvsc-mm-tuw09S .grvsc-tuw09S-3 { color: #6089B4; }
    .grvsc-mm-tuw09S .grvsc-tuw09S-15 { color: #676867; }
    .grvsc-mm-tuw09S .grvsc-tuw09S-4 { color: #9A9B99; }
  }
</style>",
    },
  ],
  "type": "root",
}
`;

exports[`prefers-color-scheme supports prefers-color-scheme with dynamically selected themes 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "lang": "js",
      "type": "html",
      "value": "<pre class=\\"grvsc-container solarized-light grvsc-mm-tlfloc\\" data-language=\\"js\\"><code class=\\"grvsc-code\\"><span class=\\"grvsc-line\\"><span class=\\"grvsc-tiaGIi-13,grvsc-tiaGIi-b grvsc-tlfloc-4\\">const</span><span class=\\"grvsc-tiaGIi-1 grvsc-tlfloc-1\\"> </span><span class=\\"grvsc-tiaGIi-7 grvsc-tlfloc-12\\">x</span><span class=\\"grvsc-tiaGIi-1 grvsc-tlfloc-1\\"> </span><span class=\\"grvsc-tiaGIi-8 grvsc-tlfloc-1\\">=</span><span class=\\"grvsc-tiaGIi-1 grvsc-tlfloc-1\\"> </span></span>
<span class=\\"grvsc-line\\"><span class=\\"grvsc-tiaGIi-3,grvsc-tiaGIi-i grvsc-tlfloc-3\\">// Comment</span></span></code></pre>",
    },
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "lang": "js",
      "type": "html",
      "value": "<pre class=\\"grvsc-container solarized-light grvsc-mm-tsSx3w\\" data-language=\\"js\\"><code class=\\"grvsc-code\\"><span class=\\"grvsc-line\\"><span class=\\"grvsc-tiaGIi-13,grvsc-tiaGIi-b grvsc-tsSx3w-9,grvsc-tsSx3w-i\\">const</span><span class=\\"grvsc-tiaGIi-1 grvsc-tsSx3w-1\\"> </span><span class=\\"grvsc-tiaGIi-7 grvsc-tsSx3w-1\\">x</span><span class=\\"grvsc-tiaGIi-1 grvsc-tsSx3w-1\\"> </span></span>
<span class=\\"grvsc-line\\"><span class=\\"grvsc-tiaGIi-3,grvsc-tiaGIi-i grvsc-tsSx3w-3\\">// Comment</span></span></code></pre>",
    },
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "type": "html",
      "value": "<style class=\\"grvsc-styles\\">
  
  .solarized-light { background-color: #FDF6E3; }
  .solarized-light .grvsc-tiaGIi-i { font-style: italic; }
  .solarized-light .grvsc-tiaGIi-b { font-weight: bold; }
  .solarized-light .grvsc-tiaGIi-13 { color: #073642; }
  .solarized-light .grvsc-tiaGIi-1 { color: #657B83; }
  .solarized-light .grvsc-tiaGIi-7 { color: #268BD2; }
  .solarized-light .grvsc-tiaGIi-8 { color: #859900; }
  .solarized-light .grvsc-tiaGIi-3 { color: #93A1A1; }
  
  /* Dark+ (default dark) */
  @media (prefers-color-scheme: dark) {
    .grvsc-mm-tlfloc {
      background-color: #1E1E1E;
      color: #D4D4D4;
    }
    .grvsc-mm-tlfloc .grvsc-tlfloc-4 { color: #569CD6; }
    .grvsc-mm-tlfloc .grvsc-tlfloc-1 { color: #D4D4D4; }
    .grvsc-mm-tlfloc .grvsc-tlfloc-12 { color: #9CDCFE; }
    .grvsc-mm-tlfloc .grvsc-tlfloc-3 { color: #6A9955; }
  }
  
  /* Monokai */
  @media (prefers-color-scheme: dark) {
    .grvsc-mm-tsSx3w {
      background-color: #272822;
      color: #f8f8f2;
    }
    .grvsc-mm-tsSx3w .grvsc-tsSx3w-i { font-style: italic; }
    .grvsc-mm-tsSx3w .grvsc-tsSx3w-9 { color: #66D9EF; }
    .grvsc-mm-tsSx3w .grvsc-tsSx3w-1 { color: #F8F8F2; }
    .grvsc-mm-tsSx3w .grvsc-tsSx3w-3 { color: #75715E; }
  }
</style>",
    },
  ],
  "type": "root",
}
`;

exports[`sets highlighted line class names 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "lang": "js{1,3-4}",
      "type": "html",
      "value": "<pre class=\\"grvsc-container default-dark\\" data-language=\\"js\\"><code class=\\"grvsc-code\\"><span class=\\"grvsc-line grvsc-highlighted\\"><span class=\\"mtk3\\">// 1</span></span>
<span class=\\"grvsc-line\\"><span class=\\"mtk3\\">// 2</span></span>
<span class=\\"grvsc-line grvsc-highlighted\\"><span class=\\"mtk3\\">// 3</span></span>
<span class=\\"grvsc-line grvsc-highlighted\\"><span class=\\"mtk3\\">// 4</span></span>
<span class=\\"grvsc-line\\"><span class=\\"mtk3\\">// 5</span></span></code></pre>",
    },
    Object {
      "children": Array [],
      "type": "paragraph",
    },
    Object {
      "type": "html",
      "value": "<style class=\\"grvsc-styles\\">
  
  .default-dark {
    background-color: #1E1E1E;
    color: #D4D4D4;
  }
  .default-dark .mtk3 { color: #6A9955; }
</style>",
    },
  ],
  "type": "root",
}
`;
